### PR TITLE
Fix/clang tidy new

### PR DESCRIPTION
**Pull request by notzain**

### CODE REVIEW

File: engine/events/SDLEventHandler.h [hunk 1]
```diff
@@ -1,14 +1,17 @@
 #pragma once
 
 #include "IEventHandler.h"
-#include "input/SDLKeys.h"
+
+#include <engine/input/SDLKeys.h>
 #include <vector>
 
+union SDL_Event;
```
Comment (FabioGNR) [inline engine/events/SDLEventHandler.h]: what are you doing C++

File: engine/game/IGame.cpp [hunk 2]
```diff
@@ -17,9 +17,9 @@ void IGame::previous()
     }
 }
 
-void IGame::next(std::shared_ptr<engine::State> state)
+void IGame::next(const std::shared_ptr<engine::State>& state)
 {
-    if(!m_states.empty()) {
+    if (!m_states.empty()) {
         auto currentState = m_states.top();
```
Comment (FabioGNR) [inline engine/game/IGame.cpp]: currentState should be reference right? was my commit lost?

File: game/level/levelReader/LevelReader.cpp [hunk 3]
```diff
@@ -2,12 +2,12 @@
 #include <fstream>
```
Comment (FabioGNR) [inline game/level/levelReader/LevelReader.cpp]: was this file removed or is it now duplicated? game/level/reader/LevelReader.cpp?

File: tests/engine/ecs/ecs_test.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            auto& entity = world.createEntity();
+            world.addComponent<TestComponent>(entity, TestComponent{});
+        }
+
+        std::size_t iteratedEntities = 0;
+        world.forEachEntityWith<TestComponent>([&](const auto& /*entity*/) {
+            iteratedEntities++;
+        });
+
+        REQUIRE(entityCount == iteratedEntities);
```
Comment (FabioGNR) [inline tests/engine/ecs/ecs_test.cpp]: This test doesn't really prove that only entities with that component are looped, maybe it's worth adding a few more entities without the component

File: game/level/levelReader/LevelReader.cpp (hunk 3 above)
Comment (FabioGNR) [reply game/level/levelReader/LevelReader.cpp]: image
I see this in PR changes, but can't find the second file (level/reader/LevelReader.cpp) locally when I pull ¯\_(ツ)_/¯

Comment (FabioGNR) [reply game/level/levelReader/LevelReader.cpp]: Nevermind, was still in develop branch. game/level/levelReader/LevelReader.cpp should be deleted as there is also a game/level/reader/LevelReader.cpp

File: engine/events/SDLEventHandler.h (hunk 1 above)
Comment (notzain) [reply engine/events/SDLEventHandler.h]: forward declaring a C-api :^)

File: engine/game/IGame.cpp (hunk 2 above)
Comment (notzain) [reply engine/game/IGame.cpp]: looks like somewhere, some commits got lost? have to look into it

File: tests/engine/ecs/ecs_test.cpp (hunk 4 above)
Comment (notzain) [reply tests/engine/ecs/ecs_test.cpp]: truu dat

**Comment by FabioGNR** [comment]
Thanks, if you remove duplicate file 
image I will approve!

**Pull request closed without merging**